lms/djangoapps/instructor/views/api.py

```diff
@@ -1031,6 +1031,36 @@ def get_students_features(request, course_id, csv=False):  # pylint: disable=red
 
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def get_students_who_may_enroll(request, course_id):
+    """
+    Initiate generation of a CSV file containing information about
+    students who may enroll in a course.
+
+    Responds with JSON
+        {"status": "... status message ..."}
+
+    """
+    course_key = CourseKey.from_string(course_id)
+    query_features = ['email']
+    try:
+        instructor_task.api.submit_calculate_may_enroll_csv(request, course_key, query_features)
+        success_status = _(
+            "Your students who may enroll report is being generated! "
```

Comment:
@mhoeber This is a breaking change.

It affects the following test: [`test_calculate_report_csv_success`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor/tests/test_api.py#L2276)

The test breaks because it requires success messages for CSV generation tasks to conform to the following format:

    Your {report_type} report is being generated! 
    You can view the status of the generation task in the 'Pending Instructor Tasks' section.

Currently, all of the CSV generation tasks produce success messages that conform to this format:

* Grade reports:

        Your grade report is being generated!
        You can view the status of the generation task in the 'Pending Instructor Tasks' section.

* Profile information for enrolled students:

        Your enrolled student profile report is being generated! 
        You can view the status of the generation task in the 'Pending Instructor Tasks' section.

* List of students who may enroll:

        Your students who may enroll report is being generated! 
        You can view the status of the generation task in the 'Pending Instructor Tasks' section.

We could of course relax the test to make sure it still passes after applying the changes you requested. However, I think it would be preferable to keep enforcing consistency between success messages of different tasks. One way of achieving this would be to change the success messages for *all* of the CSV generation tasks.

CC @sarina